Fixes to DFG

oxisto opened the pull request:
# New Nodes
+ MemoryAddress: (nearly) every node has a memoryAddress now, for which we track the current value stored there. For Declarations, we create a new memoryAddress, the References use the same memoryAddress as their refersTo
+ UnknownMemoryValue: The value stored might be unknown, in which case this node comes in
+ ParameterMemoryValue: The value that comes in via a parameter. 
+ PointerReference (&a): Has an input (in this case Reference „a“), and indicates that we actually use the Reference (as in C, so the memory address, not as in CPG-Reference) of the input
+ PointerDereference (*p): Has an input (in this case Reference „p“), and indicates that we actually use the Dereference, as in the value we find when using the input as address

# FunctionSummaries
  What does the function do? What’s the new value of arguments after the function is finished? 
Mostly for internal processing. Allows us to process each function only once and then process the affects of function calls

# New (DFG)-Edges
+ currentDeref(Deref)Value: Mostly relevant for Expression/Pointerhandling. I.e, if we have „int i=0; int* p=&i“, then the currentDerefValue of „p“ points to „0“.
+ shortFS: Gives the DFG-Summaries-Edge that does not enter other functions, but only indicates that i.e. an argument is handed to a function and that the function does something with it. Handy if we want to travers larger DFGs and don’t want to enter every single function
+ memoryValue: The current value stored for a Node, mostly relevant for Expressions. I.e. for „i=0; printf(i)“, the 2nd Reference „i“ should have a memoryValue-Edge pointing to the Literal 0.

KuechA commented on cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/concepts/crypto/hashing/Hashing.kt:
```diff
@@ -0,0 +1,77 @@
+import de.fraunhofer.aisec.cpg.graph.concepts.crypto.IsCryptography
+import java.util.Objects
+
+enum class HashingAlgorithm(val algorithmName: String) {
```
Probably not required any more

KuechA committed:
Update cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/functional/BasicLatticesRedesign.kt

KuechA committed:
Merge branch 'main' into ak/try-to-fix-merge-points

morbitzer committed:
processed review

morbitzer committed:
processed review

KuechA committed:
Merge branch 'main' into ak/try-to-fix-merge-points

KuechA committed:
higher timeout

morbitzer committed:
merge new fixpoint iteration

morbitzer committed:
spotless

morbitzer committed:
comment next broken test

morbitzer committed:
optimized EOG iteration

morbitzer committed:
also add the new labels to the edges between nodes

morbitzer committed:
be more stable on edge selection

morbitzer committed:
spotless

morbitzer committed:
Merge branch 'main' into small-dfg-fix

morbitzer committed:
adapted nestedLoop test to new behavior

morbitzer committed:
consider paralell EOG-Edges

morbitzer committed:
added analyze timeout

morbitzer committed:
uncommented test

morbitzer committed:
renamed timeout

morbitzer committed:
compare both states

morbitzer committed:
fixed loop sensitivity

morbitzer committed:
fixed testReturnStatement()

morbitzer committed:
remove debug code

morbitzer committed:
merged main

morbitzer committed:
uncommented two tests

morbitzer committed:
continue on non-empty sccEdgesQueue

morbitzer committed:
re-activted testMultipleValues()

morbitzer committed:
DFG-Edges from FieldDeclarations to unwritten MemberExpressions

morbitzer committed:
declare dummy FS

morbitzer committed:
re-enabled statementTest

renovate[bot] committed:
Update dependency com.github.javaparser:javaparser-symbol-solver-core to v3.28.0 (#2530)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency eslint-plugin-svelte to v3.14.0 (#2531)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

morbitzer committed:
started drawing dfg-edges via parameterdeclarations

morbitzer committed:
started drawing dfg-edges via parameterdeclarations II

morbitzer committed:
started drawing dfg-edges via parameterdeclarations III

morbitzer committed:
finished drawing dfg-edges via parameterdeclarations

morbitzer committed:
adapted testMatching

morbitzer committed:
adapted testPropagateArguments

morbitzer committed:
re-activated ShortcutsTest

morbitzer committed:
re-enabled testUnresolvedCalls

morbitzer committed:
re-activated queryTest

morbitzer committed:
re-activated policytest

morbitzer committed:
support for TupleDeclarations

morbitzer committed:
re-activated testFunctionLevelTuple()

morbitzer committed:
re-enable go task

morbitzer committed:
re-enable go task II

morbitzer committed:
writes to memberExpressions are now also the lastwrites to their base

morbitzer committed:
fixed testSimpleFieldDataflow

morbitzer committed:
nits

renovate[bot] committed:
Update dependency @sveltejs/kit to v2.49.5 [SECURITY] (#2533)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

morbitzer committed:
consider all EOGStarterHolders

morbitzer committed:
fixed memorytest

morbitzer committed:
re-enabled most loggingconcepttest

KuechA committed:
Add functionality to the MCP server to call specific passes (#2472)

* Make it easier to configure the server from somewhere else

* Start to add pass stuff to the MCP server

* Fix compile errors in test

* fix compile errors

* Add tool running the passes - ordering is missing

* register tool, fix test

* Fix integration tests

* Enforcing required passes

* Identify suitable targets for the passes

* Provide some tests

* Test the error handling

* Some cleanup

* Even more fine-grained tracking of already executed passes per node to avoid unnecessary work

* Update description

* Remove default argument

* Apply suggestions from code review

Co-authored-by: Maximilian Kaul <maximilian.kaul@aisec.fraunhofer.de>

* Review part 1

* Formatting

* Review part 2

* Move descriptions to the classes

* Note in the docs

* More review

* Find language-specific passes by name which makes them optional

* Also account for replacing passes per language

---------

Co-authored-by: Maximilian Kaul <maximilian.kaul@aisec.fraunhofer.de>

KuechA committed:
Cleanup of warnings after a year of coding (#2534)

* Reduce compiler warnings in cpg-core

* cpg-mcp cleanup

* Cleanup cpg-neo4j

* Cleanup of codyze-console

* Cleanup of codyze-core

* Cleanup of cpg-analysis

* Cleanup cpg-analysis

* Cleanup of cpg-concepts

* Cleanup of cpg-language-jvm

* Cleanup of cpg-language-java

* Cleanup of cpg-language-python

* Cleanup of tests

* Suppress some warnings

renovate[bot] committed:
Update dependency @sveltejs/kit to v2.50.0 (#2535)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update jackson monorepo to v2.21.0 (#2536)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

morbitzer committed:
fetch values for pointerReferences via getAddresses

morbitzer committed:
check for FS before calling acceptInternal

morbitzer committed:
soft dependency on dfg passes

morbitzer committed:
soft dependency on dfg passes II

morbitzer committed:
use provide evaluator to evaluate()

maximiliankaul committed:
Fix CDG for ShortCircuitOperators (#2502)

* Try to improve identifying merge points in iterateEOG

* Connect the BB nodes with the graph

* Add scc class

* Improve computation of region

* Improve toString of basic block

* Fix test

* Change SCC to an EOG edge property

* initial SCC support

* first version for support for nested SCCs

* better version for support for nested SCCs + Test

* iterateEOG: prioritize SCC Labeled EOG Edges

* better handling of mergePoints

* fixed most tests

* cleaned up

* Fix neo4j test

* basic block comments and refactoring

* Fix

* Some refactoring of SccPass

* added more complex EOG

* sccPass: consider single-BB SCCs

* Add tests, ...

* Fix CDG for ShortCircuitOperators

* Some fixes

* Minor cleanup

* Less code

---------

Co-authored-by: Alexander Kuechler <alexander.kuechler@aisec.fraunhofer.de>
Co-authored-by: Mathias Morbitzer <mathias.morbitzer@aisec.fraunhofer.de>
Co-authored-by: KuechA <31155350+KuechA@users.noreply.github.com>

morbitzer committed:
less verbose output from handleCallExpression

morbitzer committed:
better timeout

maximiliankaul committed:
Make Neo4j `pushToNeo4j` more easily available. (#2540)

* Make Neo4j `pushToNeo4j` more easily available.

This commit moves connection default values to a new file and provides a
new function to persist a TranslationResult without already having a
session available.

* remove old function and rename new to pushToNeo4j

* fix tests to use the new function

* make CLI arguments be used again

KuechA committed:
Fix MCP Server bugs (#2538)

* Fix typo in run_pass tool

* Fix name of nodeId to be more consistent

* Fix test

renovate[bot] committed:
Update dependency com.github.ajalt.clikt:clikt to v5.1.0 (#2541)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

KuechA committed:
Try to fix QueryTree IDs (#2537)

* Try to fix QueryTree IDs

* Fix ID in test

* Fix bug in querytree id calculation and update

* Javadoc

morbitzer committed:
better output

morbitzer committed:
started on functionpointer support

morbitzer committed:
re-activated more tests

morbitzer committed:
re-activated walkertets

morbitzer committed:
remove println()

lshala committed:
Python: Add implicit constructor return and fix receiver scoping (#2544)

* Fix DFG for Python constructors

* Fix changes

* Also consider return as change in state

* Fix symbol resolving of receiver

* Move addDecleration outside the receiver default value condition

* Remove workaround and extend existing test

---------

Co-authored-by: Alexander Kuechler <alexander.kuechler@aisec.fraunhofer.de>

morbitzer committed:
better timeout handling

morbitzer committed:
added timeout flag to iterateEOG

renovate[bot] committed:
Update dependency io.ktor.plugin to v3.4.0 (#2547)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency org.jboss.forge.roaster:roaster-api to v2.31.0.Final (#2548)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency com.vanniktech:gradle-maven-publish-plugin to v0.36.0 (#2542)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: KuechA <31155350+KuechA@users.noreply.github.com>

morbitzer committed:
fix broken CDG

morbitzer committed:
progress on functionPointers

morbitzer committed:
progress on functionPointers II

morbitzer committed:
re-activated 2nd FunctionPointer test

morbitzer committed:
fix DynamicLoadingTest

morbitzer committed:
re-activated testFunctionCallOrTypeCast

morbitzer committed:
fixed testParenthesis()

KuechA committed:
Upgrade to mcp-sdk v0.8.3 and remove deprecation warnings (#2549)

maximiliankaul committed:
Jep link error: refere to the README (#2552)

fwendland committed:
Apply quick-fixes and fix typos (#2522)

* Apply quick-fixes and fix typos for code quality and reduced compiler warnings

* Address comments
- extend comment
- change comment into @suppress

* Spotless

morbitzer committed:
fixed CXXLanguageFrontendTest

morbitzer committed:
s/ControlFlowSensitiveDFGPass/PointsToPass/g

morbitzer committed:
additional DFG edges for MethodDeclarations

morbitzer committed:
fixed testUnresolvedCallsWithInference()

KuechA committed:
Fix cdg timeout test

KuechA committed:
Merge main into small-dfg-fix

KuechA committed:
Compare once, not twice

morbitzer committed:
handleprevDFG: only follow full DFGs

morbitzer committed:
avoid DFGPass doing handleInitializerListExpression(node)

morbitzer committed:
Revert "avoid DFGPass doing handleInitializerListExpression(node)"

This reverts commit adc4cf574ef819d2980b4fa94351764ca9c7eda9.

KuechA committed:
Higher timeout for tests

morbitzer committed:
also analyze first node in eog

KuechA committed:
Merge branch 'main' into small-dfg-fix

morbitzer committed:
add functionname as parent to PMVs

morbitzer committed:
make mapDstToSrc Identity-based

morbitzer committed:
fixed tests, replaced some HashMaps

morbitzer committed:
less verbose ValueEvaluator

morbitzer committed:
fixed test

KuechA committed:
Merge branch 'main' into small-dfg-fix

morbitzer committed:
re-activated testFollowFunctions

morbitzer committed:
attempt to add loop detection to valueEvaluator

morbitzer committed:
added handling of InitializerListExpression

morbitzer committed:
attempt to add loop detection to valueEvaluator II

morbitzer committed:
handleCallExpression: only analyze invokes-edges that are not the empty declaration

morbitzer committed:
re-activated handleSubscriptExpression()

morbitzer committed:
analyze limit for InitializerListExpressions

morbitzer committed:
fixed testDictAccess

morbitzer committed:
attempt to add loop detection to valueEvaluator III

morbitzer committed:
attempt to add loop detection to valueEvaluator IV

morbitzer committed:
propagate newly to create memoryAddresses through functionSummaries

morbitzer committed:
remove ValueEvaluator in isNullCheck

morbitzer committed:
propagate newly to create memoryAddresses through functionSummaries II

morbitzer committed:
improved PMV DFG handling

KuechA committed:
Merge branch main into small-dfg-fix

KuechA committed:
More fixes

renovate[bot] committed:
Update dependency svelte to v5.53.5 [SECURITY] (#2610)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency org.mockito:mockito-core to v5.22.0 (#2613)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency svelte-check to v4.4.4 (#2614)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

morbitzer committed:
give binaryOps a memoryAddress

morbitzer committed:
better DFGTest

morbitzer committed:
fix additional dfg edges

morbitzer committed:
ignored testPropagateArguments

renovate[bot] committed:
Update dependency typescript-eslint to v8.56.1 (#2621)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

lshala committed:
Python: Fix self-assigned fields and or-expression type propagation (#2602)

* Add second test

* Fix failing test and adjust binary operator handling

* Add assert

* Handle or and and operators via assignedTypes

* Refactor default implementation of propagateAssignedTypes

lshala committed:
Upgrades the Kotlin MCP SDK from 0.8.4 to 0.9.0  (#2623)

* Upgrade mcp sdk to 0.9.0

* First try with ChannelTransport

* Add cancel and close in finally block

* Try more tests

* Try ListCommandsTest

* Use new test function in ApplyConceptsTest

* Fix remaining tests

* Add join() to wait until cancel is done

* Fix imports

* Revert changes back

* Refactor test setup

* Add missing functionality from mcp-sdk-testing example

* Get rid of unnecessary functions

* Add comment

* Replace runTest with runBlocking to use the ChannelTransport's default dispatcher

* Replace runTest in all remaining tests

* Use CompletableDeferred as in the example

* Remove unnecessary launch/join for client.connect

* Try tests with timeout

* Refactor test setup again

* Refactor McpTestSetup

* Add unconfined dispatcher for channel transport

* Remove runBlocking in tests

* Refactor setup

* Add comment

* Fix imports

* Refactor test setup

konradweiss committed:
Kw/statements as expressions (#2619)

* Renaming parent folder

* Adapting paths in file contents

* Moving expression from expressions.expressions to expressions

* Adapt file and package paths for expressions moved up

* Add usedAsExpression as a generic field in all statements and expression, the default is true, because the unifying node will be the expression, subclasses can set the default deifferently

* Adding expression fields, inheritance and impls to statement

* Removing original expression node

* Replace statement with expression

* Propagating replacement to usages in files

* Fixing tests and failed match by string

* Adding previous statements to dfg construction not only because they are expressions, but also if they are marked as being used as an expression

* Add DFG Handling of Statement nodes and fix issue in fluent dsl

* Reset removing coditional and Ifelse unification, keeping implementation of dfgpass handlers for statements now expressions

* Correct import paths

* Using magic flag to fix test for fluentdsl

* Spotless

* Fixes

* Reset IfElse addArgument functionallity

* Undo some block addition changes and other changes to fluent

* spotless

* Adapt fqn in persisted concepts yaml

* Address some of the review comments

* Removed remaining unneded FQN usages

* Remove unnecessary FQN Usages that this PR added unintentionally

* Adapt usedAsExpression to be a constructor parameter only used in the supertype declaration to define the behavior and value for the different expressiontypes

* Spotless

* Decoupling DFG addition for breakable nodes, blocks and declarations from the EOG construction

* Remove propagation of usedAsExpression to subexpression and adds some tests

* Add more tests

* Spotless

* Testfix
# Conflicts:
#	cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/analysis/abstracteval/value/IntegerValue.kt
#	cpg-concepts/src/test/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/TagOverlaysPassTest.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/evaluation/MultiValueEvaluator.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/evaluation/SizeEvaluator.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Interfaces.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/declarations/Function.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/declarations/Variable.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/expressions/Expression.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/expressions/ForEach.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/expressions/MemoryAddress.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/expressions/PointerDereference.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/expressions/PointerReference.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/helpers/functional/BasicLatticesRedesign.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/ControlDependenceGraphPass.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/ControlFlowSensitiveDFGPass.kt
#	cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/DFGPass.kt
#	cpg-core/src/test/kotlin/de/fraunhofer/aisec/cpg/enhancements/DFGFunctionSummariesTest.kt
#	cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/CXXDeclarationTest.kt
#	cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/CXXLanguageFrontendTest.kt
#	cpg-language-llvm/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/llvm/StatementHandler.kt
#	cpg-language-llvm/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/llvm/StatementHandlerTest.kt
#	cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/DFGTest.kt
#	cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonValueEvaluatorTest.kt

renovate[bot] committed:
Update dependency @sveltejs/kit to v2.55.0 (#2625)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency org.mockito:mockito-core to v5.23.0 (#2626)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

morbitzer committed:
consider casts in expressionhandler

morbitzer committed:
more accurate isGlobal for Pointer(De)References

fwendland committed:
Prevent potential division by zero when calculating 'time per LoC' (#2627)

* Prevent potential division by zero when calculating 'time per LoC'

* Simplify code using `max(...)`

morbitzer committed:
better argument-parameter matching

lshala committed:
MCP Server: Replace the `cpg_llm_analyze` tool with a `suggest_concepts` MCP prompt.  (#2631)

Replace llm_analyze_tool with suggest_concepts_prompt

morbitzer committed:
Analyze all eogStarterHolders

lshala committed:
MCP Server: Add `CpgDfgBackwardTool` (#2632)

Add CpgDfgBackwardTool

morbitzer committed:
updated ignore on testPropagateArgumentsControlFlowInsensitive()

konradweiss committed:
Not persisting result of root in type which created several root edges (#2637)

Not persisting result of root in type which created several root edges between the same nodes

morbitzer committed:
consider writes to global Variables in Assigns

morbitzer committed:
consider writes to global Variables in Assigns II

morbitzer committed:
activated most fileconcepttests

morbitzer committed:
temporary isGlobal hack for contextmanager

morbitzer committed:
add dependson PtP

morbitzer committed:
reactivated testArithmetic

morbitzer committed:
avoid analyzing eogstarters multiple times

KuechA committed:
Non-local variables pass

morbitzer committed:
replaced isGlobal hack for contextmanager

morbitzer committed:
reenabled go integration test

morbitzer committed:
reenabled bct

morbitzer committed:
reenabled GoLanguageFrontendTest

morbitzer committed:
reenabled GoLanguageFrontendTest II

morbitzer committed:
reenabled typetest

morbitzer committed:
reenabled integrationtests

morbitzer committed:
reenabled CodyzeScriptHostTest

KuechA committed:
Fix things

KuechA committed:
Merge main into small-dfg-fix

KuechA committed:
Infer more functions in case of problematic empty result

KuechA committed:
Remove duplicated tests

KuechA committed:
Ignore tests which fail since merge

KuechA committed:
Ignore also integration tests

morbitzer committed:
consider initializerLists in assignment's lhs

morbitzer committed:
correcting handling of multiple destAddresses in updateValues

morbitzer committed:
reenabled testMkstempMkdtemp

# Conflicts:
#	cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/file/python/PythonFileConceptPass.kt

morbitzer committed:
import fix

morbitzer committed:
reenabled 2 more tests

morbitzer committed:
consider new renamings

morbitzer committed:
got testContextSensitive() to work

morbitzer committed:
handle forEach

morbitzer committed:
fixed broken tests

morbitzer committed:
fixed testSignatureMatch

morbitzer committed:
fixed testChainedCall()

morbitzer committed:
fixed testUnwrapReference

morbitzer committed:
fix testMemberAccess

morbitzer committed:
re-enabled expressiontest

morbitzer committed:
fixed testPropagateArguments()

morbitzer committed:
fixed testPythonLoadIni

KuechA committed:
Workaround

morbitzer committed:
support for comprehensions

morbitzer committed:
fix testNestedFieldFlow()

morbitzer committed:
reenabled 2 tests

morbitzer committed:
reenabled compileMacOSX8664

KuechA committed:
Merge branch 'main' into small-dfg-fix

morbitzer committed:
add info for partialTargets

morbitzer committed:
removed currentDerefValues

morbitzer committed:
avoid drawing unnecessary dfg-edges between params

morbitzer committed:
draw DFG edges from PMV also when src is no param

oxisto committed:
Add output to testTraversal()

oxisto committed:
Using TRACE for PointsToPass debugging

morbitzer committed:
fixed broken test

morbitzer committed:
fix error on empty constantExpression

morbitzer committed:
started on testFunctionSummaryDerefs

morbitzer committed:
spotless

KuechA committed:
Extend function summaries with "function" origin

KuechA committed:
Merge branch 'main' into small-dfg-fix

KuechA committed:
Spotless

KuechA committed:
Fixes

morbitzer committed:
continued on testFunctionSummaryDerefs

morbitzer committed:
fix broken tests

KuechA committed:
Node paths now also include the edges

KuechA committed:
Also provide the edge path in the QueryTree

morbitzer committed:
added testMemberArguments

morbitzer committed:
consider writes to memberAccesses in calls

morbitzer committed:
fixed MemberAccessTest

morbitzer committed:
overhauled realcode test

morbitzer committed:
moved more tests to their own testfile

morbitzer committed:
increated testTraversal timeout

KuechA committed:
Merge branch 'main' into small-dfg-fix

morbitzer committed:
started on testStructReference

morbitzer committed:
fixed broken tests

morbitzer committed:
added testfile

morbitzer committed:
improved handling of arrays

morbitzer committed:
cleaned up I

morbitzer committed:
exchanged concurrentIS for IS

morbitzer committed:
removed some debug info

morbitzer committed:
exchanged concurrentIS for IS II

morbitzer committed:
consider initializerLists that contain only assigns

KuechA committed:
try to fix delete statements

KuechA committed:
Add file

KuechA committed:
Extend test

morbitzer committed:
handling arrayInitializers that are constructors

morbitzer committed:
reworked delete handling

morbitzer committed:
improved handling of arrayinitializer

KuechA committed:
Try some fix

KuechA committed:
revert parts of b0451d14aef0066c0eafab0030f33560b3f9ca14

KuechA committed:
Fix test

KuechA committed:
Revert things back again

morbitzer committed:
exchanged concurrentIS for IS III

morbitzer committed:
exchanged concurrentIS for IS IV

morbitzer committed:
increased chunk size

morbitzer committed:
introduced forEachMaybeParallel

morbitzer committed:
introduced forEachMaybeParallel II

morbitzer committed:
introduced forEachMaybeParallel III

morbitzer committed:
introduced forEachMaybeParallel IV

morbitzer committed:
removed asyncs

morbitzer committed:
more cleanup

maximiliankaul committed:
Make IniFileFrontend open

KuechA reviewed (approved):
Let's finally get rid of this

KuechA committed:
Enable/disable points to pass

oxisto committed:
Removed unncessary log line

oxisto committed:
Fixed failing tests

oxisto committed:
dont add pass excplicitly

oxisto committed:
Merge branch 'main' into small-dfg-fix
